pkg/cgroups/cgroupblkio.go

```diff
@@ -0,0 +1,168 @@
+func GetBlkioDir() string {
+	return blkioCgroupDir
+}
+
+// SetBlkioParameters writes OCI BlockIO parameters to files in cgroups blkio contoller directory.
+func SetBlkioParameters(cgroupsDir string, blockIO OciBlockIOParameters) error {
+	log.Debug("configuring cgroups blkio controller in directory %#v with parameters %+v", cgroupsDir, blockIO)
+	var errors *multierror.Error
+	if blockIO.Weight >= 0 {
+		errors = multierror.Append(errors, writeToFileInDir(cgroupsDir, blkioWeightFiles, strconv.FormatInt(blockIO.Weight, 10)))
+	}
+	for _, weightDevice := range blockIO.WeightDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioWeightDeviceFiles, weightDevice.Major, weightDevice.Minor, weightDevice.Weight))
+	}
+	for _, rateDevice := range blockIO.ThrottleReadBpsDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioThrottleReadBpsFiles, rateDevice.Major, rateDevice.Minor, rateDevice.Rate))
+	}
+	for _, rateDevice := range blockIO.ThrottleWriteBpsDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioThrottleWriteBpsFiles, rateDevice.Major, rateDevice.Minor, rateDevice.Rate))
+	}
+	for _, rateDevice := range blockIO.ThrottleReadIOPSDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioThrottleReadIOPSFiles, rateDevice.Major, rateDevice.Minor, rateDevice.Rate))
+	}
+	for _, rateDevice := range blockIO.ThrottleWriteIOPSDevice {
+		errors = multierror.Append(errors, writeDevValueToFileInDir(cgroupsDir, blkioThrottleWriteIOPSFiles, rateDevice.Major, rateDevice.Minor, rateDevice.Rate))
+	}
+	return errors.ErrorOrNil()
+}
+
+// writeDevValueToFileInDir writes MAJOR:MINOR VALUE to the first existing file under baseDir
+func writeDevValueToFileInDir(baseDir string, filenames []string, major, minor, value int64) error {
+	content := fmt.Sprintf("%d:%d %d", major, minor, value)
+	return writeToFileInDir(baseDir, filenames, content)
+}
+
+// writeToFileInDir writes content to the first existing file in the list under baseDir.
+func writeToFileInDir(baseDir string, filenames []string, content string) error {
+	var errors *multierror.Error
+	// Returns list of errors from writes, list of single error due to all filenames missing or nil on success.
+	for _, filename := range filenames {
+		filepath := baseDir + "/" + filename
+		err := writeToFile(filepath, content)
+		if err == nil {
+			return nil
+		}
+		errors = multierror.Append(errors, err)
+	}
+	err := errors.ErrorOrNil()
+	if err != nil {
+		return fmt.Errorf("could not write content %#v to any of files \"%s\": %w", content, strings.Join(filenames, "\", \""), err)
+	}
+	return nil
+}
+
+// writeToFile writes content to an existing file.
+func writeToFile(filename string, content string) error {
+	f, err := os.OpenFile(filename, os.O_WRONLY, 0666)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, fperr := fmt.Fprintf(f, content)
```

Comment:
`_, err = fmt.Fprintf`
no need to allocate one more variable if you already have one `err`.

Reply:
Thanks, changed that to what you're suggesting. I often like single-assignment-style, but here with errors it doesn't really help anybody. :)